### PR TITLE
Add and update API documents

### PR DESCRIPTION
**Pull request by hachi8833**
Mainly added doc to  Hash, and updated some others.

### CODE REVIEW

File: vm/hash.go [hunk 1]
```diff
@@ -127,6 +167,18 @@ var builtinHashInstanceMethods = []*BuiltInMethodObject{
...
+		//
+		// ```Ruby
+		// h = { a: 1, b: "2", c: [1, 2, 3], d: { k: 'v' } }
+		// h['a'] = 1          #=> 1
+		// h['b'] = "2"        #=> "2"
+		// h['c'] = [1, 2, 3]  #=> [1, 2, 3]
+		// h['d'] = { k: 'v' } #=> { k: 'v' }
+		// ```
+		//
+		// @return [value]
```
Comment (adlerhsieh) [inline vm/hash.go]: The name in the brackets is the class of the returned value, so it should be `[Object]` instead. The `value` can go to explanation afterwards.

File: vm/hash.go [hunk 2]
```diff
@@ -152,6 +204,14 @@ var builtinHashInstanceMethods = []*BuiltInMethodObject{
 		},
 	},
 	{
+		// Return the number of key-value pairs of the hash.
```
Comment (adlerhsieh) [inline vm/hash.go]: Probably make it "Returns"?

File: vm/range.go [hunk 3]
```diff
@@ -67,6 +101,19 @@ var builtInRangeInstanceMethods = []*BuiltInMethodObject{
...
+		// ```ruby
+		// (1..5).to_a     # => [1, 2, 3, 4, 5]
+		// (1..5).to_a[2]  # => 3
+		// ```
+		//
+		// **Note:**
+		// - Only `do`-`end` block is supported for now: `{ }` block is unavailable.
+		// - Three-dot range `...` is not supported yet.
+		//
+		// @return [nil]
```
Comment (adlerhsieh) [inline vm/range.go]: Should be capitalized class name `Null`.

File: vm/string.go [hunk 4]
```diff
@@ -17,6 +17,22 @@ type RString struct {
...
+// ```ruby
+// a = "Three"
+// b = 'zero'
+// c = '漢'
+// d = 'Tiếng Việt'
+// e = "😏️️"
+// ```
+//
+// **Note:**
+// Currently, manipulations are based upon Golang's Unicode manipulations.
```
Comment (adlerhsieh) [inline vm/string.go]: Add a blank line between the sentence and the bullet points.

File: vm/hash.go (hunk 1 above)
Comment (hachi8833) [inline vm/hash.go]: Fixed

File: vm/hash.go (hunk 2 above)
Comment (hachi8833) [inline vm/hash.go]: Fixed

File: vm/range.go (hunk 3 above)
Comment (hachi8833) [inline vm/range.go]: Fixed

File: vm/string.go (hunk 4 above)
Comment (hachi8833) [inline vm/string.go]: Fixed

**Comment by st0012** [comment]
@hachi8833 Nice work!

**Pull request merged**